distribution metrics guide (#9014)
DataDog/documentation: content/en/tracing/guide/ddsketch_trace_metrics.md

  tag: "Documentation"
  text: "Learn more about Distributions"
- link: "https://docs.datadoghq.com/tracing/guide/metrics_namespace/"
  tag: "Documentation"
  text: "Learn more about Trace Metrics"
---

Trace metrics are collected automatically for your services and resources and retained for 15 months. The latency percentiles exist as individual time series and in beta as a DD Distribution Metric. This means that we now offer more data in easier formats. Instead of having a different metric for each percentile, separately for services, resources and second primary tags, we now offer a simple metric:

- `trace.<SPAN_NAME>`:
  - *Prerequisite:* This metric exists for any APM service in beta.
  - *Description:* Represents latency distributions for all services, resources and versions across different environments and second primary tags.
  - *Metric type:* [DISTRIBUTION][1]
  - *Tags:* `env`, `service`, `version`, `resource`, and [the second primary tag][2].

The APM Service and Resource pages now use this new metric for customers in the Beta automatically. This means you can use these metrics to power your dashboards and monitors.

**What does this metric mean for me?**

> - Distribution metrics offer new capabilities compared with the existing gauge metric like precise aggregation in time (instead of a rollup) and open the door for better visualizations and functionality in the future.

Reviewer: - Distribution metrics offer new capabilities compared with the existing gauge metric like precise percentile calculation over time periods (instead of a rollup function) and open the door for better visualizations and functionality in the future.

---
title: DDSketch based Metrics in APM
kind: guide
aliases:
- /tracing/faq/resource-trace-doesn-t-show-up-under-correct-service/
further_reading:
- link: "https://www.datadoghq.com/blog/engineering/computing-accurate-percentiles-with-ddsketch/"
  tag: "Blog"
  text: "Computing Accurate Percentiles with DDSketch"
- link: "https://docs.datadoghq.com/metrics/distributions/"
  tag: "Documentation"
  text: "Learn more about Distributions"
- link: "https://docs.datadoghq.com/tracing/guide/metrics_namespace/"
  tag: "Documentation"
  text: "Learn more about Trace Metrics"
---

> 

Reviewer 2: Since a lot of this guide talks about something in beta, let's put the standard alert box up here, something like...

```
<div class="alert alert-warning">
DDSketch trace metrics are in beta. Request access by...
 </div>
```

- `trace.<SPAN_NAME>`:
  - *Prerequisite:* This metric exists for any APM service in beta.
  - *Description:* Represents latency distributions for all services, resources and versions across different environments and second primary tags.
  - *Metric type:* [DISTRIBUTION][3]
  - *Tags:* `env`, `service`, `version`, `resource`, and [the second primary tag][2].

The APM Service and Resource pages now use this metric type for customers in the Beta automatically. This means you can use these metrics to power your dashboards and monitors.

**What does this metric mean for me?**
- [Distribution metrics][1] offer new capabilities like precise percentile calculation over time periods (instead of a rollup function) and open the door for better visualizations and functionality in the future.

**How am I seeing a full history of this new metric?**

> - We stitch any existing query on the new metric to an equivalent query based on the long-existing latency metrics, so you never have to create multiple queries

Reviewer 2: ```suggestion
- Datadog stitches any existing query on the new metric to an equivalent query based on the long-existing latency metrics, so you do not have to create multiple queries.
```

**I’m seeing a change in the values of my latency, what is happening?**
- Datadog Distribution Metrics are powered by [DDSketch][4]. This includes a change from rank-error guarantees to relative error guarantees. As a result, all percentile estimate values are now guaranteed to be closer to the true percentile value.

> - Specifically you might expect to see a reduction in p99 values where this difference is most noticeable. The new values are centered more closely on the precise p99 value.

Reviewer 2: ```suggestion
- Specifically, you might expect to see a reduction in p99 values, where this difference is most noticeable. The new values are centered more closely on the precise p99 value.
```

> - One thing to note is that the APM metric calculations are not exactly akin to a DD Distribution Custom Metric that would be calculated in-code. The calculation happens on the backend so some differences may occur

Reviewer 2: ```suggestion
- One thing to note is that the APM metric calculations are not exactly akin to a DD Distribution Custom Metric that would be calculated in-code. The calculation happens on the backend, so some differences may occur.
```

**What does this beta program entail?**

> - Distribution Metrics now power APM Service and Resource pages as we roll this out for more and more customers

Reviewer 2: ```suggestion
- Distribution Metrics now power APM Service and Resource pages as Datadog rolls this out for more customers.
```

> - We will migrate all manual queries from dashboards and monitors to the new metrics automatically on your behalf while retaining the existing metrics to support historic views

Reviewer 2: ```suggestion
- Datadog will migrate all manual queries from dashboards and monitors to the new metrics automatically on your behalf, while retaining the existing metrics to support historical views.
```

The APM Service and Resource pages use this metric type for customers in the Beta automatically. This means you can use these metrics to power your dashboards and monitors.

**What does this metric mean for me?**
- [Distribution metrics][1] offer new capabilities like precise percentile calculation over time periods (instead of a rollup function) and open the door for better visualizations and functionality in the future.

**How am I seeing a full history of this new metric?**
- Datadog stitches any existing query on the new metric to an equivalent query based on the long-existing latency metrics, so you do not have to create multiple queries.

**I’m seeing a change in the values of my latency, what is happening?**
- Datadog Distribution Metrics are powered by [DDSketch][4]. This includes a change from rank-error guarantees to relative error guarantees. As a result, all percentile estimate values are now guaranteed to be closer to the true percentile value.
- Specifically, you might expect to see a reduction in p99 values, where this difference is most noticeable. The new values are centered more closely on the precise p99 value.

> - One thing to note is that the APM metric calculations are not exactly akin to a DD Distribution Custom Metric that would be calculated in-code. The calculation happens on the backend, so some differences may occur.

Reviewer 2: ```suggestion
- One thing to note is that the APM metric calculations are not exactly akin to a Datadog Distribution Custom Metric that would be calculated in-code. The calculation happens on the backend, so some differences may occur.
```
